seq_generator.py

```diff
@@ -84,13 +87,26 @@ def inc_repeats():
     result["symbols"] = grammar.symbols
     return result
 
-def create_dataset(min_length, max_length, num_samples, loc_grammar, loc_dataset):
+def create_dataset(min_length, max_train_length, max_test_length, num_train_samples, num_id_samples,
+                   num_ood_samples, loc_grammar, loc_dataset):
     grammar = ruleset.Ruleset()
     grammar.load(loc_grammar)
 
-    r = generate_data(min_length, max_length, num_samples, grammar)
+    train_data = generate_data(min_length, max_train_length, num_train_samples, grammar, "train")
     with open(loc_dataset, "w") as f:
-        json.dump(r, f, indent=4)
+        json.dump(train_data, f, indent=4)
+
+    non_terminals = list(grammar.rules.keys())
+
+    # omit unproductive rules from test sets
+    for nt in non_terminals:
+        for rule in grammar.rules[nt]:
+            if (not rule[1]):
+                grammar.rules[nt].remove(rule)
+    
+    # not sure how to export/save the data yet, will be completed later
```

Comment:
You can use the generate_data function (also in seq_generator). It is on the main branch.

Reply:
I apologize but I am not quite sure what you mean. The create_dataset function which I modified still makes calls to the existing generate_data method. Could you please elaborate a bit further?

Reply:
Sorry I confused myself when reading the code. What I meant is that you can use the same approach to export the data as the create_data function on main does. (Basically just dump whatever generate_data returns as json)